x-pack/plugins/upgrade_assistant/server/routes/cluster_checkup.ts

```diff
@@ -20,15 +22,30 @@ export function registerClusterCheckupRoutes({ cloud, router }: RouteDependencie
       async (
         {
           core: {
+            savedObjects: { client: savedObjectsClient },
             elasticsearch: { dataClient },
           },
         },
         request,
         response
       ) => {
         try {
+          const status = await getUpgradeAssistantStatus(dataClient, isCloudEnabled);
+
+          const callAsCurrentUser = dataClient.callAsCurrentUser.bind(dataClient);
+          const reindexActions = reindexActionsFactory(savedObjectsClient, callAsCurrentUser);
+          const reindexService = reindexServiceFactory(
+            callAsCurrentUser,
+            reindexActions,
+            log,
+            licensing
+          );
+          const indexNames = status.indices.map(index => index.index);
```

Comment:
Could we add a `.filter(Boolean)` or similar after the `.map` to remove any potential undefined values? That way `cleanupReindexOperations` can accept `string[]` instead of `Array<string | undefined>`?

Reply:
Great point! Fixed.